doc/developer/scan.txt

```diff
@@ -132,22 +62,262 @@ files, and section of the code they deal with, are :
+  scan operator.
+
+
+Notation
+========
+
+Scan being a sizeable and complex module, it has its own naming convention for
+functions and variables which this section will attempt to introduce.
+
+Recall that a Scan op contains a Theano function representing the computation
+that is done in a single iteration of the loop represented by the Scan op (in
+other words, the computation given by the function provided as value to
+theano.scan's ''fn'' argument ). Whenever we discuss a scan op, "outer function"
+refers to the Theano function that *contains* the Scan op whereas the "inner
+function" refers to the Theano function that is *contained* in the Scan op.
+
+In the same spirit, the inputs to the Apply node wrapping the Scan op (or "Scan
+node" for short) and its outputs are referred to as "outer inputs" and "outer
+outputs", respectively, because these inputs and outputs are variables in the
+outer function graph. The inputs and outputs of scan's inner function are
+designated "inner inputs" and "inner outputs".
+
+
+Scan variables
+==============
+
+The following are the different types of variables that Scan has the capacity to
+handle, along with their various caracteristics.
+
+**Sequence** : A sequence is a Theano variable which Scan will iterate over and
+give sub-elements to its inner function. For a sequence variable `X`, at
+timestep `t`, the inner function will receive as input the sequence element
+`X[t]`. These variables are used through the argument `sequences` of the
+`theano.scan()` function.
+
+**Non-sequences** : A non-sequence is a Theano variable which Scan will provide
+`as-is` to its inner function. For a non-sequence variable X, at timestep `t`,
+the inner function will receive as input the variable X. These variables are
+used through the argument `non_sequences` of the `theano.scan()` function.
+
+**Nitsot (no input tap, single output tap)** : A nitsot is an output variable of
+the inner function that is not fed back as an input to the next iteration of the
+inner function. Nitsots are typically encountered in situations where Scan is
+used to perform a 'map' operation (every element in a tensor is independently
+altered using a given operation to produce a new tensor) such as squaring every
+number in a vector.
+
+**Sitsot (single input tap, single output tap)** : A sitsot is an output
+variable of the inner function that is fed back as an input to the next
+iteration of the inner function. A typical setting where a sitsot might be
+encountered is the case where scan is used  Sitsots are typically encountered in
+situations where Scan is used to sum the elements of a vector and a sitsot
+output is employed to act as an accumulator.
+
+**Mitsot (multiple input taps, single output tap)** : A mitsot is an output
+variable of the inner function that is fed back as an input to future iterations
+of the inner function (either multiple future iterations or a single one that
+isn't the immediate next one).
```

Comment:
You could mention the Fibonacci sequence as an example: F(n) gets used to compute both F(n+1) and F(n+2).